exchanges/stream/websocket.go

```diff
@@ -1274,6 +1275,10 @@ func (w *Websocket) GetConnection(messageFilter any) (Connection, error) {
 		return nil, errMessageFilterNotSet
 	}
 
+	if request.IsMockResponse(ctx) {
+		return newMockWebsocketConnection(), nil
```

Comment:
This placement does not appear the most appropriate. GetConnection creating a whole new one while existing functionality does not? 
Why not have a new function instead of this? Reduces the need for inspecting context on every call as well. Even if you don't make a new function, I think it should be removed from this one
I don't see any benefit to this over just assigning a websocket with `stream.MockWebsocketConnection` inside test code